runtime/test/planner-tests.js

```diff
@@ -951,6 +951,9 @@ view MyThings1 of [Thing1] #mythings1 in 'things.json'
 view MyThings2 of [Thing1] #mythings2 in 'things.json'`);
 
     // Two transformation particle hosting the same particle with different type storage.
+    // NOTE: This doesn't work yet because we don't have a way of representing a concrete
+    // type with type variable'd handles.
+    /*
```

Comment:
Can you add a test case where one is trying to connect [Thing2] handle to the Muxer with P1 hosted particle and it doesn't resolve? Basically, verify the over-permissiveness being fixed?